lib/vector-config-macros/src/configurable.rs

```diff
@@ -35,8 +35,7 @@ pub fn derive_configurable_impl(input: TokenStream) -> TokenStream {
         let where_clause = generics.make_where_clause();
         for typ in generic_field_types {
             let ty = &typ.ident;
-            let predicate =
-                parse_quote! { #ty: ::vector_config::Configurable + ::serde::Serialize };
+            let predicate = parse_quote! { #ty: ::vector_config::Configurable + ::serde::Serialize + ::vector_config::ToValue };
```

Comment:
Hmmm, I think we could just drop the bound on `Serialize` here and add a generic impl of `ToValue for T where T: Serialize` near the `ToValue` trait declaration.

This would let us avoid the below boilerplate implementation of `ToValue` that requires us to then re-export `serde_json`, etc.

Reply:
I was hoping to avoid a generic impl, for… reasons, but I guess it doesn't really matter. It certainly shrinks up the code requirements.

Reply:
😂 

Any reason in particular? I'm not against boilerplate... in this case, it's almost entirely write-once-never-think-about-it-again, but I'm mostly curious if there's something about doing a blanket implementation that hurts us elsewhere, or otherwise pigeonholes us.

Reply:
The biggest one was a nagging worry (likely unjustified?) that dragging in `Serialize` would make it no longer object safe somehow. The efficiency angle is there too (dragging in a trip through serialization just to produce `Value::Number` is gross that way) but that's not _really_ an issue for schema generation.

Reply:
Yeah, all reasonable and fair.

We can leave it as-is for now, because doing my suggestion is trivial to accomplish in the future during a regular ol' refactor/cleanup pass.